src/guiguts/maintext.py

```diff
@@ -703,6 +703,102 @@ def find_matches(
                 start += f"+{count_var.get()}c"
         return matches
 
+    def transform_selection(self, fn) -> None:
+        """Transform a text selection by applying a function or method.
+
+        Args:
+            fn: Reference to a function or method
+        """
+        sel = self.selected_ranges()
+        if not sel:
+            # nothing is currently selected
+            return
+
+        p1 = self.index("sel.first")
```

Comment:
At the moment, this doesn't cope well if the user does a "column" (or block) selection (`Option+drag mouse` on Mac).
For each feature we will need to decide whether (or when) to implement support for column selections. 
Internally, column selections consist of multiple selections (normally between column M and N on multiple rows to form a rectangle). If we decide not to support column selections for a feature, then the best  option is probably to just work on the first selection, i.e. the first line of the block. That is what I have done for the "Count" feature in the Search dialog for now (though I would envisage adding column selection support at some point). If we do want to support column selection (like GG1 does for case-changing) then we need to loop on the list of selections that `selected_ranges()` returns. An example of this is in `column_copy_cut`.

Reply:
Aha, so that's why it's possible to return multiple ranges... I'm not accustomed to using column selection so that was off my radar. I'll revisit this and see if I can loop over a set of them.

Reply:
I only became accustomed to it because of GG1 development, though I use it quite a lot in Notepad++ when editing code.

Reply:
I also changed from the prior behavior and not re-selecting the text after modification. First, because it wasn't working. And second because GG1 doesn't do that.

Reply:
> Added support for column selections. @windymilla, if you wouldn't mind a quick re-test...

Seems to work well, thanks.

I noticed that you had duplicated my code from `column_copy_cut` to handle the special "case where line was too short to select any text", and had two thoughts:
1. We probably don't want to have to do that everywhere we use column selections
2. In this specific case it doesn't matter since changing the case of a newline character does nothing anyway.

So I've added a separate issue #139 which one of us can pick up after this PR is merged. For now, you can either remove or leave the special case code in `transform_selection`.